### ISSUE TITLE
Import user defined functions in Python

### ISSUE THREAD

**Issue by luboss** [sqlitebrowser/sqlitebrowser#2262]
Describe the new feature
--------------------------
I would like to define specific functions in Python which I could import to DB4S and use them in queries. Just like in [sqlite3.Connection.create_function](https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.create_function).
This is probably way beyond the scope of DB4S but perhaps it could be achieved by developing an extension (following https://docs.python.org/3/extending/building.html). It would open many new possibilities. I could work on it but might need some little help when in doubts. I never did this before.

Additional info
---------------

#### Is your feature request related to an issue? Please include the issue number.
No

#### Does this feature exist in another product or project? Please provide a link.
Basically yes - [sqlite3.Connection.create_function](https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.create_function)

#### Do you have a screenshot? Please add screenshots to help explain your idea.
I use example from the link I've posted. Considering I as user write little simple Python code which I want to use in the queries inside DB browser:
```
def md5sum(t):
    return hashlib.md5(t).hexdigest()
```
Import it (or better have an interface where I store my functions) and then just use the new function in my quries:
```
SELECT MD5SUM(username) FROM users;
```

**Comment by scottfurry** [comment]
We currently do not have any reliance on the python sqlite module ([API Docs here](https://docs.python.org/3/library/sqlite3.html)). The code makes usage of the C++ QT libraries so there would need to be some interface to shiboken2 _at a minimum_ for python function usage.

I'm not sure of the group as a whole of the inclination of the amount of work involved for python function usage. It is not a small undertaking.

**Comment by justinclift** [comment]
It's an interesting concept.  Not sure who'd be up for creating it though. :man_shrugging:

**Comment by MKleusberg** [comment]
Sorry for the long and unorganised comment :wink: Just some thoughts here.

SQLite allows you to register new functions for usage in SQL queries. Additionally it allows you to load extension files which can contain one or more of these functions. The ```sqlite3.Connection.create_function``` function looks like it registers a single new function written in Python.

We have an interface for loading extension files in DB4S. You can find that in the Preferences dialog or by clicking Tools -> Load Extension. This loads .dll/.so files, so you would need to find a way to create those with Python. Just registering a single function, like you do, is way more complicated for us because we would need to open and check the file with the function, find the entry point of the function, and then register it - which is exactly what SQLite would be doing for us when loading an extension. And when it's just a Python file we would need to include a Python interpreter too.

So I would think the easiest way to achieve what you want is to see if it's possible to write a C/C++ .dll/.so which behaves like an SQLite extension (so we can load it in DB4S) and which does nothing more than forwarding the function calls to a Python script. When written as some sore of generic template we could include it in our source tree too :smile:
